sunpy/image/resample.py

```diff
@@ -82,31 +82,44 @@ def resample(orig, dimensions, method='linear', center=False, minusone=False):
 def _resample_nearest_linear(orig, dimensions, method, offset, m1):
     """
     Resample Map using either linear or nearest interpolation.
+
+    Parameters
+    ----------
+    orig :
+        Original data.
+    dimensions :
+        Dimensions of resamled data.
+    offset :
+        Either 0 or 0.5, depending on whether interpolation is at the edge or
+        centers of bins.
+    m1 :
+        For ``orig.shape = (i,j)`` & new dimensions ``= (x,y)``, if set to `False`
+        (default) ``orig`` is resampled by factors of ``(i/x) * (j/y)``,
+        otherwise ``orig`` is resampled by ``(i-1)/(x-1) * (j-1)/(y-1)``.
+        This prevents extrapolation one element beyond bounds of input array.
     """
     dimlist = []
 
-    # calculate new dims
-    for i in range(orig.ndim):
-        base = np.arange(dimensions[i])
-        dimlist.append((orig.shape[i] - m1) / (dimensions[i] - m1) *
-                       (base + offset) - offset)
-
-    # specify old coordinates
-    old_coords = [np.arange(i, dtype=np.float) for i in orig.shape]
+    # Calculate old coords. Add 0.5 to get centers of pixels
```

Comment:
I am a little confused by this comment. You are adding `offset` right, which could be 0.5 or not?

Reply:
Yep, old comment